extensions/resource-deployment/src/ui/deployAzureSQLVMWizard/pages/sqlServerSettingsPage.ts

```diff
@@ -0,0 +1,242 @@
+						},
+						{
+							component: this._sqlAuthenticationPasswordConfirmationTextRow
+						}
+						// {
+						// 	component: this.wizard.createFormRowComponent(view, 'SQL Storage Optimization Type', '', this._sqlStorageOptimiazationDropdown, true)
+
+						// }
+					],
+					{
+						horizontal: false,
+						componentWidth: '100%'
+					})
+				.withLayout({ width: '100%' })
+				.component();
+
+			return view.initializeModel(this._form);
+		});
+	}
+
+	public async onEnter(): Promise<void> {
+		this.wizard.wizardObject.registerNavigationValidator((pcInfo) => {
+			if (pcInfo.newPage < pcInfo.lastPage) {
+				return true;
+			}
+
+			let username = this._sqlAuthenticationTextbox.value!;
+
+			let showErrorMessage = '';
+
+			if (username.length < 2 || username.length > 128) {
+				showErrorMessage += 'Username must be between 2 and 128 characters long.\n';
+			}
+
+			if (/[!@#$%^&*()_+\-=\[\]{};':"\\|,.<>\/?]+/.test(username)) {
+				showErrorMessage += 'Username cannot contain special characters \/""[]:|<>+=;,?* .\n';
+			}
+
+			showErrorMessage += this.wizard.validatePassword(this._sqlAuthenticationPasswordTextbox.value!);
+
+			if (this._sqlAuthenticationPasswordTextbox.value !== this._sqlAuthenticationPasswordConfirmationTextbox.value) {
+				showErrorMessage += ('Password and confirm password must match.');
+				return false;
+			}
+
+			this.wizard.showErrorMessage(showErrorMessage);
+
+			if (showErrorMessage !== '') {
+				return false;
+			}
+			return true;
+		});
+	}
+
+	public onLeave(): void {
+		this.wizard.wizardObject.registerNavigationValidator((pcInfo) => {
+			return true;
+		});
+	}
+
+	private createSqlConnectivityDropdown(view: azdata.ModelView) {
+		this._sqlConnectivityDropdown = view.modelBuilder.dropDown().withProperties(<azdata.DropDownProperties>
+			{
+				values: [
+					{
+						name: 'local',
+						displayName: 'Local (inside VM only)'
+					},
+					{
+						name: 'private',
+						displayName: 'Private (within Virtual Network)'
+					},
+					{
+						name: 'public',
+						displayName: 'Public (Internet)'
+					}
+				],
+				value: {
+					name: 'private',
+					displayName: 'Private (within Virtual Network)'
+				}
+			}).component();
+
+		this.wizard.model.sqlConnectivityType = (this._sqlConnectivityDropdown.value as azdata.CategoryValue).name;
+
+		this._sqlConnectivityDropdown.onValueChanged((value) => {
+
+			let connectivityValue = (this._sqlConnectivityDropdown.value as azdata.CategoryValue).name;
+			this.wizard.model.sqlConnectivityType = connectivityValue;
+
+			if (connectivityValue === 'local') {
+				this.wizard.changeRowDisplay(this._portTextRow, 'none');
+			} else {
+				this.wizard.changeRowDisplay(this._portTextRow, 'block');
+			}
+		});
+
+	}
+
+	private createPortText(view: azdata.ModelView) {
+		this._portTextBox = view.modelBuilder.inputBox().withProperties(<azdata.InputBoxProperties>{
+			inputType: 'number',
+			max: 65535,
+			min: 1024,
+			value: '1433'
```

Comment:
Hmm, if we're accepting essentially numbers for min and max, wouldn't it make sense to change the signature to string | number for value?